server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1592,6 +1601,9 @@ private SearchSourceBuilder parseXContent(XContentParser parser, boolean checkTr
                 throw new ParsingException(parser.getTokenLocation(), "Unexpected token [" + token + "] found after the main object.");
             }
         }
+        if (retrieverBuilder != null) {
+            retrieverBuilder.extractToSearchSourceBuilder(this);
+        }
```

Comment:
FYI, I am not sure about this. It seems to be SearchSourceBuilder should have a bunch of retrievers it runs. Extracting like this seems like an anti-pattern.

Reply:
I agree with your thoughts here, and this is not intended to be the end design. I'm trying to chunk off as little as possible, so it made sense to me to do it this way as the first step to get the parsing code in. Next step I'll make it so the `SearchSourceBuilder` is rebuilt as a retriever and that is passed to the shard.